### PR TITLE
Stef/natgrads notebook revamp

### PR DESCRIPTION
**Pull request by stefanosele**
Cleaned up natgrad notebook and better documented it.

* Remove dependency on the Action framework
* Now all optimizations in the notebook are done by creating a `.make_optimize_tensor()` and
manually run it through a session. That should be more consistent with the GPflow 2.0 way of optimizing a model.
* Deliberately removed `run_nat_grads_with_adam()` and `run_adam()`. It leads to some code duplication for each particular case we are covering, but  I believe is much closer to how the user is going to use it (I can revert that bit back if you believe that we should avoid the duplication)

### CODE REVIEW

File: doc/source/notebooks/manual/advanced/natural_gradients.ipynb [hunk 1]
```diff
@@ -4,15 +4,881 @@
...
+   ],
+   "source": [
+    "vgp_bernoulli_natgrad.compute_log_likelihood()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also choose to run natural gradients in another parameterization.</br>\n",
```
Comment (vdutor) [inline doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: The `</br>` didn't render correctly.

File: doc/source/notebooks/manual/advanced/natural_gradients.ipynb [hunk 2]
```diff
@@ -4,15 +4,881 @@
...
+   "execution_count": 48,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2018-06-13T13:31:56.149218Z",
+     "start_time": "2018-06-13T13:31:55.213980Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "from matplotlib import pyplot as plt\n",
```
Comment (vdutor) [inline doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: `from matplotlib import pyplot as plt`
`import tensorflow as tf`
aren't used.

File: doc/source/notebooks/manual/advanced/natural_gradients.ipynb [hunk 3]
```diff
@@ -4,15 +4,881 @@
...
+   ],
+   "source": [
+    "gpr.compute_log_likelihood()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we will create an approximate model which approximates the true posterior via a variational Gaussian distribution. <br\\> We initiallize the distribution to be zero mean and unit variance."
```
[the remark this answers is not in the document]
Comment (awav) [reply doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: <br\\> => </br>

File: doc/source/notebooks/manual/advanced/natural_gradients.ipynb [hunk 4]
```diff
@@ -4,15 +4,881 @@
...
+      "GPR with Adam: iteration 1 likelihood -230.6706\n",
+      "GPR with Adam: iteration 2 likelihood -230.2508\n",
+      "GPR with Adam: iteration 3 likelihood -229.8303\n",
+      "GPR with Adam: iteration 4 likelihood -229.4093\n",
+      "GPR with Adam: iteration 5 likelihood -228.9876\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i in range(iterations):\n",
```
[the remark this answers is not in the document]
Comment (awav) [reply doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: it is 5 iterations

File: doc/source/notebooks/manual/advanced/natural_gradients.ipynb (hunk 1 above)
Comment (awav) [reply doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: What do you mean?

File: doc/source/notebooks/manual/advanced/natural_gradients.ipynb (hunk 4 above)
[the remark this answers is not in the document]
Comment (vdutor) [reply doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: how long do each of them take?

File: doc/source/notebooks/manual/advanced/natural_gradients.ipynb (hunk 1 above)
Comment (vdutor) [reply doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: if I open the notebook file, Their is no line break, I can just see the `<br>`

File: doc/source/notebooks/manual/advanced/natural_gradients.ipynb (hunk 4 above)
[the remark this answers is not in the document]
Comment (awav) [reply doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: I did this `iterations = notebook_niter(5)`. I do not know how long does it take, different hardware - different time.

File: doc/source/notebooks/manual/advanced/natural_gradients.ipynb (hunk 1 above)
Comment (awav) [reply doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: How do you open it? View file gives normal line break. I can't see anything.

Comment (vdutor) [reply doc/source/notebooks/manual/advanced/natural_gradients.ipynb]: Chrome webbrowser. It looks fine now - maybe the page didn't load completely.

**Pull request merged**